src/NzbDrone.Core/Statistics/StatisticsService.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NzbDrone.Core.Datastore;
+using NzbDrone.Core.History;
+using NzbDrone.Common.Extensions;
+
+namespace NzbDrone.Core.Statistics
+{
+    public interface IStatisticsService
+    {
+        StatisticsGrouping<HistoryStatistics> GetGlobalStatistics();
+        Dictionary<string, StatisticsGrouping<HistoryStatistics>> GetIndexerStatistics();
+    }
+
+    public class StatisticsService : IStatisticsService
+    {
+        private readonly IHistoryRepository _historyRepository;
+
+        public StatisticsService(IHistoryRepository historyRepository)
+        {
+            _historyRepository = historyRepository;
+        }
+
+        public StatisticsGrouping<HistoryStatistics> GetGlobalStatistics()
+        {
+            var stats = new StatisticsGrouping<HistoryStatistics>();
+
+            foreach (var group in GetIndexerStatistics().Values)
+            {
+                stats.LastWeek.Grabs += group.LastWeek.Grabs;
+                stats.LastWeek.Replaced += group.LastWeek.Replaced;
+                stats.LastWeek.Failed += group.LastWeek.Failed;
+                stats.LastWeek.Imported += group.LastWeek.Imported;
+
+                stats.LastMonth.Grabs += group.LastMonth.Grabs;
+                stats.LastMonth.Replaced += group.LastMonth.Replaced;
+                stats.LastMonth.Failed += group.LastMonth.Failed;
+                stats.LastMonth.Imported += group.LastMonth.Imported;
+
+                stats.AllTime.Grabs += group.AllTime.Grabs;
+                stats.AllTime.Replaced += group.AllTime.Replaced;
+                stats.AllTime.Failed += group.AllTime.Failed;
+                stats.AllTime.Imported += group.AllTime.Imported;
+            }
+
+            return stats;
+        }
+
+        public Dictionary<string, StatisticsGrouping<HistoryStatistics>> GetIndexerStatistics()
+        {
+            var all = _historyRepository.All().ToArray();
```

Comment:
this could get extremely slow. we should be able to run pretty much all of this right on the DB.